File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/ECommerceWebsite/OrderDetailsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderDetailsRow } from "./OrderDetailsRow";

export interface OrderDetailsColumns {
    Id: Column<OrderDetailsRow>;
    UserName: Column<OrderDetailsRow>;
    ProductName: Column<OrderDetailsRow>;
    Quantity: Column<OrderDetailsRow>;
    OrderName: Column<OrderDetailsRow>;
    Price: Column<OrderDetailsRow>;
}

export class OrderDetailsColumns extends ColumnsBase<OrderDetailsRow> {
    static readonly columnsKey = 'ECommerceWebsite.OrderDetails';
    static readonly Fields = fieldsProxy<OrderDetailsColumns>();
}

File: AdminPanel/AdminPanel.Web/Modules/ECommerceWebsite/OrderDetails/OrderDetailsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminPanel.ECommerceWebsite.Columns;

[ColumnsScript("ECommerceWebsite.OrderDetails")]
[BasedOnRow(typeof(OrderDetailsRow), CheckNames = true)]
public class OrderDetailsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string UserName { get; set; }
    public string ProductName { get; set; }
    public int Quantity { get; set; }
    public string OrderName { get; set; }
    public decimal Price { get; set; }
}